include/rocksdb/file_system.h

```diff
@@ -112,14 +112,19 @@ struct IOOptions {
   // fsync, set this to force the fsync
   bool force_dir_fsync;
 
+  // Can be used by underlying file systems to skip recursing through sub
+  // directories and list only files in GetChildren API.
+  bool do_not_recurse;
```

Comment:
This is a bad name. GetChildren is never recursive. It never returns children of children, only direct children. The question is whether to filter out directories (which would preclude wrapping GetChildren in recursive listing logic, but the name here is "do not recurse").

Also, AFAIK for backward compatibility and generality, this is only intended as a hint, not a strong requirement to filter out directories.

My `GetChildrenForFiles()` suggestion avoids extra "bloat" on IOOptions (fields only affecting one operation) and should make it easier to "do the right thing" rather than require an explicit IOOptions every time you want just the files in a directory. (See possible omission I pointed out in your code.) Also avoids the problem of "negative naming" on boolean variables. See e.g. https://www.serendipidata.com/posts/naming-guidelines-for-boolean-variables